### PR TITLE
feature: ECA-10 order list page

### PR DESCRIPTION
**Pull request by valkandyba**

### CODE REVIEW

File: src/components/OrderTable/OrderTable.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+function priceRow(qty: number, price: number) {
+  return qty * price;
+}
+
+function total(items: readonly Row[]) {
+  return items
+    .map(({ qty, price }) => priceRow(qty, price))
+    .reduce((sum, i) => sum + i, 0);
+}
```
Comment (ValeriyaZavozova) [inline src/components/OrderTable/OrderTable.tsx]: Name of function should start with a verb (get, calculate, set, find, etc)

Code style suggestion

```suggestion
function calculateItemTotal(item: Row) {
  const { qty, price } = item;
  return qty * price;
}

function calculateOrderTotal(items: readonly Row[]) {
  return items
    .map(calculateItemTotal)
    .reduce((total, amount) => total + amount, 0);
}
```

File: src/components/OrderTable/OrderTable.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  TableHead,
+  TableRow,
+} from '@mui/material';
+import classes from './OrderTable.module.scss';
+
+interface OrderTableProps {
+  rows: Row[];
+}
+
+interface Row {
```
Comment (ValeriyaZavozova) [inline src/components/OrderTable/OrderTable.tsx]: It's not clear what is `Row`
```suggestion
interface OrderItem {
```

File: src/components/OrderTable/OrderTable.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from '@mui/material';
+import classes from './OrderTable.module.scss';
+
+interface OrderTableProps {
+  rows: Row[];
```
Comment (ValeriyaZavozova) [inline src/components/OrderTable/OrderTable.tsx]: It's not clear what is rows. Think not in terms of layout but in terms of data that should be here.
Layout can be changed from table to grid and you will have to change variable too.

```suggestion
  orderItems: OrderItem[];
```

File: src/components/OrderTable/OrderTable.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+function total(items: readonly Row[]) {
+  return items
+    .map(({ qty, price }) => priceRow(qty, price))
+    .reduce((sum, i) => sum + i, 0);
+}
+
+const OrderTable: React.FC<OrderTableProps> = ({ rows }) => {
+  const invoiceTotal = total(rows);
+  const invoiceSubtotal = (invoiceTotal / (TAX_RATE * 100 + 100)) * 100;
+  const invoiceTaxes = invoiceTotal - invoiceSubtotal;
```
Comment (ValeriyaZavozova) [inline src/components/OrderTable/OrderTable.tsx]: I think that should be more clear
```suggestion
  const invoiceTaxes = invoiceTotal * TAX_RATE;
  const invoiceSubtotal = invoiceTotal - invoiceTaxes;
```

File: src/components/OrderTable/OrderTable.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+interface Row {
+  name: string;
+  img: string;
+  qty: number;
+  price: number;
+}
+
+const TAX_RATE = 0.2;
+
+function ccyFormat(num: number) {
```
Comment (ValeriyaZavozova) [inline src/components/OrderTable/OrderTable.tsx]: Name of the function is not clear

File: src/components/OrderTable/OrderTable.tsx (hunk 1 above)
Comment (valkandyba) [reply src/components/OrderTable/OrderTable.tsx]: fixed

File: src/components/OrderTable/OrderTable.tsx (hunk 2 above)
Comment (valkandyba) [reply src/components/OrderTable/OrderTable.tsx]: fixed

File: src/components/OrderTable/OrderTable.tsx (hunk 3 above)
Comment (valkandyba) [reply src/components/OrderTable/OrderTable.tsx]: fixed

File: src/components/OrderTable/OrderTable.tsx (hunk 5 above)
Comment (valkandyba) [reply src/components/OrderTable/OrderTable.tsx]: fixed

File: src/components/OrderTable/OrderTable.tsx (hunk 4 above)
Comment (valkandyba) [reply src/components/OrderTable/OrderTable.tsx]: fixed

**Pull request merged**